src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -43,16 +60,15 @@ public static IMvcCoreBuilder AddMvcCore(this IServiceCollection services)
                 throw new ArgumentNullException(nameof(services));
             }
 
-            ConfigureDefaultServices(services);
-
-            AddMvcCoreServices(services);
-
-            if (setupAction != null)
+            if (setupAction == null)
```

Comment:
Why change this to `throw` when passed a `null` delegate? The previous code seemed more usable, allowing users to call the full overload all the time if they prefer.